faiss/cppcontrib/sa_decode/Level2-inl.h

```diff
@@ -12,6 +12,16 @@
 #include <cstdint>
 
 #include <faiss/cppcontrib/detail/CoarseBitType.h>
+#include <faiss/impl/platform_macros.h>
+
+#ifdef FAISS_BIG_ENDIAN
+#define Swap2Bytes(val) ((((val) >> 8) & 0x00FF) | (((val) << 8) & 0xFF00))
+#endif
+
+#ifndef FAISS_BIG_ENDIAN
+#define FAISS_BIG_ENDIAN 0
```

Comment:
I agree, this needs a closer attention.

Reply:
How about moving the Swap2Bytes and Swap4Bytes definitions to platform-macros.h. Something like this is below. I will make the required changes in the other files. This will ensure we do not change FAISS_BIG_ENDIAN at any other stage at all. 

```
#define FAISS_BIG_ENDIAN 0
#if !defined(_MSC_VER) && \
        (defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__))
#define FAISS_BIG_ENDIAN 1
#endif
#define Swap2Bytes(val) ((((val) >> 8) & 0x00FF) | (((val) << 8) & 0xFF00))
#define Swap4Bytes(val)                                           \
    ((((val) >> 24) & 0x000000FF) | (((val) >> 8) & 0x0000FF00) | \
     (((val) << 8) & 0x00FF0000) | (((val) << 24) & 0xFF000000))
     
     
     
```

Reply:
If FAISS_BIG_ENDIAN is always defined (either as 0 or 1) then I would change

```
#ifdef FAISS_BIG_ENDIAN
```

to

```
#if FAISS_BIG_ENDIAN
```

Reply:
Yes, @ChipKerchner . I am making the change.

Reply:
@mdouze @ChipKerchner @alexanderguzhva Please check now. I hope this is okay.